### PR TITLE
Update cargo clippy to nightly version

### PR DESCRIPTION
**Pull request by anton-lisanin**
- Update rust compiler to latest version
- Update `cargo clippy` to latest version
- Fix clippy warnings

### CODE REVIEW

File: evm_loader/program/src/config.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ use crate::account_data::ACCOUNT_MAX_SIZE;
 cfg_if! {
     if #[cfg(feature = "mainnet")] {
 
-        const CHAIN_ID: u64 = 245022934;
+        const CHAIN_ID: u64 = 245_022_934;
```
Comment (mich-master) [inline evm_loader/program/src/config.rs:15]: Why did you change the number format in this place and leave the old format in other branches of cfg_if ?
I think, the number format should be the same.

Comment (anton-lisanin) [reply evm_loader/program/src/config.rs]: We don't really need to change config.rs.
I reverted this.

**Review by vakond — APPROVED** [review]

File: evm_loader/program/src/precompile_contracts.rs [hunk 2]
```diff
@@ -354,7 +354,7 @@ pub fn query_account<'a, B: AccountStorage>(
             let executable = state.query_solana_account_executable(account_address);
             if let Some(executable) = executable {
                 debug_print!("query_account executable result: {}", executable);
-                let executable: U256 = (executable as u8).into(); // pad to 32 bytes
+                let executable: U256 = if executable { U256::one() } else { U256::zero() }; // pad to 32 bytes
```
Comment (rozhkovdmitrii) [inline evm_loader/program/src/precompile_contracts.rs:357]: Wouldn't be From trait implementing better?

```rust
impl From<bool> for U256 {
    fn from(value: bool) -> Self {

    }
}
```

File: evm_loader/program/src/entrypoint.rs [hunk 3]
```diff
@@ -791,16 +791,8 @@ fn process_instruction<'a>(
...
-        }
-
-        EvmInstruction::Write |
-        EvmInstruction::Cancel |
-        EvmInstruction::Finalise |
-        EvmInstruction::CreateAccountWithSeed |
-        EvmInstruction::ExecuteTrxFromAccountDataIterative |
-        EvmInstruction::PartialCallFromRawEthereumTX |
-        EvmInstruction::Continue
-        => Err!(ProgramError::InvalidInstructionData; "Deprecated instruction"),
```
Comment (rozhkovdmitrii) [inline evm_loader/program/src/entrypoint.rs:803]: I think it would be better to keep deprecated cases and list them explicitly

File: evm_loader/cli/src/account_storage.rs [hunk 4]
```diff
@@ -220,14 +220,7 @@ impl<'a> EmulatorAccountStorage<'a> {
...
 
-                if let Some(acc) = config.rpc_client.get_account_with_commitment(&account_data.code_account, CommitmentConfig::processed()).unwrap().value {
-                    eprintln!("Account found");
-                    Some(acc)
-                }
-                else {
-                    eprintln!("Account not found");
-                    None
-                }
+                config.rpc_client.get_account_with_commitment(&account_data.code_account, CommitmentConfig::processed()).unwrap().value
```
Comment (rozhkovdmitrii) [inline evm_loader/cli/src/account_storage.rs:223]: I just interested if someone guarantees safety unwrapping and handling a `ClientError`

Comment (anton-lisanin) [reply evm_loader/cli/src/account_storage.rs:223]: No guarantees. It will panic if there was a network error.
But it does not really matter

File: evm_loader/program/src/precompile_contracts.rs (hunk 2 above)
Comment (anton-lisanin) [reply evm_loader/program/src/precompile_contracts.rs:357]: I prefer explicit expression for conversion between unrelated types instead of hiding it under generic `from`.
This is also important part of the business logic and should be expressed as clear as possible

File: evm_loader/program/src/entrypoint.rs (hunk 3 above)
Comment (anton-lisanin) [reply evm_loader/program/src/entrypoint.rs:803]: This part is a bit strange and inconsistent. Half of these cases will never happen.

Deprecated instructions already listed in the `instruction.rs`. 
Over time, there number will increase. 
And keeping hundredth deprecated instructions in this match will generate a lot of noise.

A agree about moving each case into a separate method, but it is out of scope of this pull request.
